README.md

```diff
@@ -0,0 +1,191 @@
+
+if (process.env.REACT_APP_MOCK_API) {
+  const appContext = buildMockLocalContext(window.appContext ?? mockContext);
+  // automatically redirects to the correct url
+  const searchParams = new URLSearchParams(window.location.search);
+  if (!searchParams.get('itemId')) {
+    searchParams.set('itemId', appContext.itemId);
+    window.location.search = searchParams.toString();
+  }
+  // set up the mock database
+  const database = window.Cypress ? window.database : buildDatabase(appContext);
+
+  const errors = window.apiErrors;
+  mockServer({ database, appContext, errors });
+}
+```
+
+6. Add the `ContextContext` and the `TokenContext` files in your app in `src/components/context`. It will handle the authentication and fetching the local context automatically for you. Don't forget to always mount these contexts (in `<Root/>` and `<App/>`).
+
+#### TokenContext.js
+
+```js
+import React, { createContext } from 'react';
+import PropTypes from 'prop-types';
+import qs from 'qs';
+import { hooks } from '../../config/queryClient';
+
+const TokenContext = createContext();
+
+const TokenProvider = ({ children }) => {
+  const { itemId } = qs.parse(window.location.search, {
+    ignoreQueryPrefix: true,
+  });
+  const { data, isLoading, isError } = hooks.useAuthToken(itemId);
+
+  if (isLoading) {
+    return 'loading...';
+  }
+
+  if (isError) {
+    console.error('An error occured while requesting the token.');
+  }
+
+  const value = data;
+  return (
+    <TokenContext.Provider value={value}>{children}</TokenContext.Provider>
+  );
+};
+
+TokenProvider.propTypes = {
+  children: PropTypes.node,
+};
+
+TokenProvider.defaultProps = {
+  children: null,
+};
+
+export { TokenContext, TokenProvider };
+```
+
+#### ContextContext.js
+
+```js
+import React, { createContext, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import qs from 'qs';
+import { hooks } from '../../config/queryClient';
+
+const Context = createContext();
+
+const ContextProvider = ({ children }) => {
+  const { itemId } = qs.parse(window.location.search, {
+    ignoreQueryPrefix: true,
+  });
+  const {
+    data: context,
+    isLoading,
+    isError,
+  } = hooks.useGetLocalContext(itemId);
+
+// here is a good place to change the app language
+
+  if (isLoading) {
+    return 'loading...';
+  }
+
+  if (isError) {
+    console.error('An error occured while fetching the context.');
+  }
+
+  // todo: define a context to default to  
+  const value = context ?? DEFAULT_LOCAL_CONTEXT;
+
+  return <Context.Provider value={value}>{children}</Context.Provider>;
+};
+
+ContextProvider.propTypes = {
+  children: PropTypes.node,
+};
+
+ContextProvider.defaultProps = {
+  children: null,
+};
+
+export { Context, ContextProvider };
+```
+
+7. Optionally, you can create an util file for custom hooks. For instance, the following hook fetches the data from the local context and the token context to avoid redondance.
+
+```js
+export const useAppData = () => {
```

Comment:
Should we invert the responsibilities here? I.e. e.g. let the `hooks.useAppData` hook implement the logic of checking which context / token to use? At least for common behaviours across apps, if they are all expected to implement the mock API

Reply:
I tried to do so, but for some reasons it doesn't go well with hooks, or at least how we've defined them. But I'll give a new try, it would be much better of course.